alacritty/src/config/monitor.rs

```diff
@@ -15,115 +19,139 @@ const DEBOUNCE_DELAY: Duration = Duration::from_millis(10);
+                        EventKind::Other if event.info() == Some("shutdown") => break,
+                        EventKind::Any
+                        | EventKind::Create(_)
+                        | EventKind::Modify(_)
+                        | EventKind::Other => {
+                            received_events.push(event);
+                        },
+                        _ => (),
+                    },
+                    Err(RecvTimeoutError::Timeout) => {
+                        // Go back to polling the events.
+                        debouncing_deadline = None;
+
+                        if received_events
+                            .drain(..)
+                            .flat_map(|event| event.paths.into_iter())
+                            .any(|path| paths.contains(&path))
+                        {
+                            // Always reload the primary configuration file.
+                            let event = Event::new(EventType::ConfigReload(paths[0].clone()), None);
+                            let _ = event_proxy.send_event(event);
+                        }
                     },
-                    _ => (),
-                },
-                Err(RecvTimeoutError::Timeout) => {
-                    // Go back to polling the events.
-                    debouncing_deadline = None;
-
-                    if received_events
-                        .drain(..)
-                        .flat_map(|event| event.paths.into_iter())
-                        .any(|path| paths.contains(&path))
-                    {
-                        // Always reload the primary configuration file.
-                        let event = Event::new(EventType::ConfigReload(paths[0].clone()), None);
-                        let _ = event_proxy.send_event(event);
-                    }
-                },
-                Ok(Err(err)) => {
-                    debug!("Config watcher errors: {:?}", err);
-                },
-                Err(err) => {
-                    debug!("Config watcher channel dropped unexpectedly: {}", err);
-                    break;
-                },
-            };
+                    Ok(Err(err)) => {
+                        debug!("Config watcher errors: {:?}", err);
+                    },
+                    Err(err) => {
+                        debug!("Config watcher channel dropped unexpectedly: {}", err);
+                        break;
+                    },
+                };
+            }
+        });
+
+        Some(Self { thread: join_handle, shutdown_tx: tx })
+    }
+
+    /// Synchronously shut down the monitor.
+    pub fn shutdown(self) {
+        // Request shutdown.
+        let mut event = NotifyEvent::new(EventKind::Other);
+        event = event.set_info("shutdown");
+        let _ = self.shutdown_tx.send(Ok(event));
+
+        // Wait for thread to terminate.
+        if let Err(err) = self.thread.join() {
+            warn!("config monitor shutdown failed: {err:?}");
```

Comment:
Using debug here since display isn't implemented on this error for some reason. But I don't think it would ever fail anyway.